test-suite-groovy/src/test/groovy/io/micronaut/docs/client/filter/GoogleAuthFilter.groovy

```diff
@@ -1,40 +1,34 @@
 package io.micronaut.docs.client.filter
 
-//tag::class[]
+import io.micronaut.context.BeanProvider
 import io.micronaut.context.annotation.Requires
+
+//tag::class[]
+
 import io.micronaut.context.env.Environment
-import io.micronaut.context.BeanProvider
-import io.micronaut.http.HttpResponse
 import io.micronaut.http.MutableHttpRequest
-import io.micronaut.http.annotation.Filter
+import io.micronaut.http.annotation.ClientFilter
+import io.micronaut.http.annotation.RequestFilter
 import io.micronaut.http.client.HttpClient
-import io.micronaut.http.filter.ClientFilterChain
-import io.micronaut.http.filter.HttpClientFilter
-import org.reactivestreams.Publisher
-import reactor.core.publisher.Flux
-import reactor.core.publisher.Mono
-
-import static io.micronaut.http.HttpRequest.GET
 
 @Requires(env = Environment.GOOGLE_COMPUTE)
-@Filter(patterns = "/google-auth/api/**")
-class GoogleAuthFilter implements HttpClientFilter {
+@ClientFilter(patterns = "/google-auth/api/**")
+class GoogleAuthFilter {
 
     private final BeanProvider<HttpClient> authClientProvider
 
     GoogleAuthFilter(BeanProvider<HttpClient> httpClientProvider) { // <1>
         this.authClientProvider = httpClientProvider
     }
 
-    @Override
-    Publisher<? extends HttpResponse<?>> doFilter(MutableHttpRequest<?> request,
-                                                  ClientFilterChain chain) {
-        Flux<String> token = Mono.fromCallable(() -> encodeURI(request))
-                .flatMap(authURI -> authClientProvider.get().retrieve(GET(authURI).header( // <2>
-                        "Metadata-Flavor", "Google"
-                )))
+    @RequestFilter
+    void filter(MutableHttpRequest<?> request) {
+        String authURI = encodeURI(request)
+        String token = authClientProvider.get().toBlocking().retrieve(HttpRequest.GET(authURI).header( // <2>
+                "Metadata-Flavor", "Google"
+        ))
 
-        return token.flatMap(t -> chain.proceed(request.bearerAuth(t)))
+        request.bearerAuth(token)
```

Comment:
this changes the example to encourage blocking, how would you write this filter without blocking?

Reply:
you can still use reactive (just return a publisher), but imo it shouldn't be the default in the guide (given virtual threads, and to avoid making the code more confusing)